pallets/bridge/src/lib.rs

```diff
@@ -333,11 +375,28 @@ pub mod pallet {
             Self::ensure_currency_registered(currency_id)?;
 
             let asset_id = AssetIds::<T>::get(currency_id);
-            T::Assets::transfer(asset_id, &who, &Self::account_id(), amount, false)?;
+            let Currency { external, fee, .. } = Currencies::<T>::get(asset_id);
+            let total_amount = amount.checked_add(fee).ok_or(ArithmeticError::Overflow)?;
+            if external {
+                T::Assets::burn_from(asset_id, &who, total_amount)?;
+            } else {
+                T::Assets::transfer(asset_id, &who, &Self::account_id(), total_amount, false)?;
```

Comment:
Will the relayer get the fee information from Currencies storage and reserve the fee itself?

Reply:
No, relayer only processes the transaction itself and does not make changes to the transaction information（like the fee）

Reply:
Fee will be reserved by `teleported`, if we want to get these reserved fees, we can sendAndSign `materialize`, just like the cross-chain extrinsics do

Reply:
No, relayer only processes the transaction itself and does not make changes to the transaction information. Fees will be reserved by `teleport`, if we want to get these reserved fee, we can sendAndSign the `materialize`, just like the cross-chain extrinsics do